### ISSUE TITLE
Fix discovery invection into setter or getter directly

### ISSUE THREAD

**Issue by paxhujing** [pamidur/aspect-injector#79]
```
 [Aspect(Scope.Global)]
    [Injection(typeof(WriteConsoleAttribute))]
    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
    public class WriteConsoleAttribute : Attribute
    {
        [Advice(Kind.After, Targets = Target.Setter)]
        public void WriteLine([Argument(Source.Arguments)] Object[] args)
        {
            Console.WriteLine(String.Join(';', args));
        }
    }
```

```
private Int32 _lineCount;
        public Int32 LineCount
        {
            get => _lineCount;
            [WriteConsole]
            set
            {
                if (_lineCount != value)
                {
                    _lineCount = value;
                    LineBet = value * BetPerLine;
                }
            }
        }
```

**Comment by pamidur** [comment]
In this case I assume (I can't check right now) it treats setter as a method. So you can either 
1. Mark whole property LineCount with WriteConsoleAttribute 
Or
2. Make advice target = Method.

Anyway, it is workaround. This should be fixed. Thanks for the report

**Comment by pamidur** [comment]
fixed in 2.0.1

**Comment by pamidur** [comment]
setter and getter are methods, so if you want to inject like this
```c#
            [WriteConsole]
            set
```

you need to have your advice target set to Method
```c#
[Advice(Kind.After, Targets = Target.Method)]
```
